### PR TITLE
AIK-3578 Refactor django source, removing possibility of crashes due to ASGI django

### PR DESCRIPTION
**Pull request by bitterpanda63**

### CODE REVIEW

File: aikido_zen/sources/functions/request_handler.py [hunk 1]
```diff
@@ -17,9 +17,8 @@ def request_handler(stage, status_code=0):
             if get_comms():
                 get_comms().send_data_to_bg_process("STATISTICS", {"action": "request"})
 
-            # Create a lifecycle cache
-            IPCLifecycleCache(get_current_context())
-            return None
+            if get_current_context():  # Create a lifecycle cache
```
Comment (willem-delbare) [inline aikido_zen/sources/functions/request_handler.py]: Does this do double work?

Comment (bitterpanda63) [reply aikido_zen/sources/functions/request_handler.py]: No, it caused an error, which it really should not imo

Comment (bitterpanda63) [reply aikido_zen/sources/functions/request_handler.py]: Ofc error was try-excepted

File: aikido_zen/sources/django/__init__.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    """
+    modified_django = importhook.copy_module(django)
+
+    former__get_response = copy.deepcopy(django.BaseHandler._get_response)
+
+    def aikido__get_response(self, request):  # Synchronous (WSGI)
+        run_init_stage(request)  # We do some initial request handling
+
+        if ratelimiting_middleware not in self._view_middleware:
+            # The rate limiting middleware needs to be last in the chain.
```
Comment (willem-delbare) [inline aikido_zen/sources/django/__init__.py:28]: are we called after user-defined middleware here?

File: aikido_zen/sources/django/ratelimiting_middleware.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+"""Exports ratelimiting_middleware function"""
+
+from ..functions.request_handler import request_handler
+
+
+def ratelimiting_middleware(request, *args, **kwargs):
+    """Aikido middleware that handles ratelimiting"""
```
Comment (willem-delbare) [inline aikido_zen/sources/django/ratelimiting_middleware.py]: this does more than rate limiting right? Let's rename?

File: aikido_zen/sources/django/__init__.py (hunk 2 above)
Comment (bitterpanda63) [reply aikido_zen/sources/django/__init__.py:28]: I would say this is outside of the scope of the current PR

**Pull request merged**